### PR TITLE
fix lint violations and CI errors: use rabbitmq via docker and only for integration-rabbitmq tests

### PR DESCRIPTION
**Pull request by graingert**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

## Description
@auvipy it looks like CI is failing on master for lint errors, this cleans those up a bit

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -87,8 +84,8 @@ before_install:
     - sudo apt install libcurl4-openssl-dev libssl-dev gnutls-dev
     - if [[ -v MATRIX_TOXENV ]]; then export TOXENV=${TRAVIS_PYTHON_VERSION}-${MATRIX_TOXENV}; fi; env
     - |
-          if [[ "$TOXENV" == *integration* ]]; then
-              sudo echo 'deb https://dl.bintray.com/rabbitmq-erlang/debian focal main' > /etc/apt/sources.list.d/rabbitmq-bintray.list
+          if [[ "$TOXENV" == *rabbitmq ]]; then
+              echo 'deb https://dl.bintray.com/rabbitmq-erlang/debian focal main' | sudo tee -a /etc/apt/sources.list.d/rabbitmq-bintray.list
```
Comment (graingert) [inline .travis.yml:88]: @auvipy so it turns out this was never using the bintray rabbitmq-erlang:

```
/home/travis/.travis/functions: line 110: /etc/apt/sources.list.d/rabbitmq-bintray.list: Permission denied
```

**Comment by graingert** [comment]
@auvipy can you cancel this build for me, please? https://travis-ci.org/github/celery/celery/builds/712734614

File: .travis.yml [hunk 2]
```diff
@@ -87,8 +84,9 @@ before_install:
     - sudo apt install libcurl4-openssl-dev libssl-dev gnutls-dev
     - if [[ -v MATRIX_TOXENV ]]; then export TOXENV=${TRAVIS_PYTHON_VERSION}-${MATRIX_TOXENV}; fi; env
     - |
-          if [[ "$TOXENV" == *integration* ]]; then
-              sudo echo 'deb https://dl.bintray.com/rabbitmq-erlang/debian focal main' > /etc/apt/sources.list.d/rabbitmq-bintray.list
+          if [[ "$TOXENV" == *rabbitmq ]]; then
+              echo 'deb https://dl.bintray.com/rabbitmq-erlang/debian focal erlang' | sudo tee /etc/apt/sources.list.d/rabbitmq-erlang-bintray.list
+              echo 'deb https://dl.bintray.com/rabbitmq/debian focal main' | sudo tee /etc/apt/sources.list.d/rabbitmq-bintray.list
```
Comment (graingert) [inline .travis.yml:89]:
```suggestion
              echo 'deb https://dl.bintray.com/rabbitmq/debian bionic main' | sudo tee /etc/apt/sources.list.d/rabbitmq-bintray.list
```

File: .travis.yml [hunk 3]
```diff
@@ -87,8 +84,9 @@ before_install:
     - sudo apt install libcurl4-openssl-dev libssl-dev gnutls-dev
     - if [[ -v MATRIX_TOXENV ]]; then export TOXENV=${TRAVIS_PYTHON_VERSION}-${MATRIX_TOXENV}; fi; env
     - |
-          if [[ "$TOXENV" == *integration* ]]; then
-              sudo echo 'deb https://dl.bintray.com/rabbitmq-erlang/debian focal main' > /etc/apt/sources.list.d/rabbitmq-bintray.list
+          if [[ "$TOXENV" == *rabbitmq ]]; then
```
Comment (graingert) [inline .travis.yml:87]:
```suggestion
          if [[ "$TOXENV" == *integration* ]]; then
```

**Comment by auvipy** [comment]
Only api-check is failing this time

**Comment by auvipy** [comment]
would be really grateful if you could check this out https://github.com/celery/celery/issues/6199#issuecomment-665304590

**Comment by graingert** [comment]
> would be really grateful if you could check this out [#6199 (comment)](https://github.com/celery/celery/issues/6199#issuecomment-665304590)

@auvipy hmmm you think we need?:

```
    - sudo mkdir -p /var/log/celery /var/run/celery /var/run/rabbitmq
    - sudo chown travis /var/log/celery /var/run/celery /var/run/rabbitmq
```

**Comment by auvipy** [comment]
> > would be really grateful if you could check this out [#6199 (comment)](https://github.com/celery/celery/issues/6199#issuecomment-665304590)
> 
> @auvipy hmmm you think we need?:
> 
> ```
>     - sudo mkdir -p /var/log/celery /var/run/celery /var/run/rabbitmq
>     - sudo chown travis /var/log/celery /var/run/celery /var/run/rabbitmq
> ```

not fully sure about that. that need to be investigated

**Comment by graingert** [comment]
@auvipy imho this PR is ready to go, it reduces the likelyhood of a flake failure by only using rabbitmq with integration-rabbitmq, and it fixes the lint errors.

I will continue to investigate eliminating flakey builds completely in #6254

**Comment by graingert** [comment]
@auvipy actually I lied before when I said this PR was ready, This time it is ready - and I'm not lying

**Pull request merged**